shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -1137,4 +1233,55 @@ public void release() {
             mNativeView.getFlutterJNI().unregisterTexture(id);
         }
     }
+
+    private class PointerState {
+        private long device;
+        private double dx;
+        private double dy;
+        private boolean pressed;
+
+        public PointerState(long device, double dx, double dy) {
+            this.device = device;
+            this.dx = dx;
+            this.dy = dy;
+            this.pressed = false;
+        }
+
+        public boolean isLastLocation(double dx, double dy) {
+            return this.dx == dx && this.dy == dy;
+        }
+
+        public void setLastLocation (double dx, double dy) {
+            this.dx = dx;
+            this.dy = dy;
+        }
+
+        public boolean isDown() {
+            return pressed;
+        }
+
+        public void setUp() {
+            pressed = false;
+        }
+
+        public void setDown() {
+            pressed = true;
+        }
+    }
+    private class PointerAction {
+
+        private final int pointerIndex;
+        private final int actionType;
+        private final int batched;
+
+        public PointerAction(int pointerIndex, int actionType, int batched) {
```

Comment:
I'd maybe call this FrameworkPointerEvent since this is going to be something the framework understands?